### PR TITLE
#nix generating encoding of nodes client in scale

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
Signed-off-by: Dzmitry Lahoda <dzmitry@lahoda.pro>

## Description
* generates all metadata for all relvant polkadot and composable runtime
* in generalized ways and puts them into ci/cd matrix runner

### CODE REVIEW

**Comment by seunlanlege** [comment]
This commit has already been merged to the base branch: [bb5a7a8](https://github.com/ComposableFi/composable/pull/2077/commits/bb5a7a83e250a5e1e6c7324736447bbab2c30c64)

**Comment by dzmitry-lahoda** [comment]
@seunlanlege base changes automagically by github as soon as base merged. so it is ok. thank you

**Review by blasrodri — CHANGES_REQUESTED** [review]

File: code/centauri/default.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{ pkgs, packages, crane, common-attrs, common-deps }: {
+
+  hyperspace-template = { relaychainHostA ? "127.0.0.1", relaychainPortA ? 9944
+    , parachainHostA ? "127.0.0.1", parachainPortA ? 9988, paraIdA ? 2001
+    , commitmentPrefixesA ? "0x6962632f", clientIdA ? "11-beefy-0"
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply code/centauri/default.nix:5]: its just refactoring i am doing now. so

@obsessed-cake  may to sure that our default docker for hyperspace and docker compose for bridge stuff use grandpa by default (see parameters provided by David in chat)

[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply code/centauri/default.nix]:
```
    , parachainHostA ? "127.0.0.1", parachainPortA ? 9988, paraIdA ? 2001
    , commitmentPrefixesA ? "0x6962632f", clientIdA ? "11-beefy-0"

    , relaychainHostB ? "127.0.0.1", relaychainPortB ? 9944
    , parachainHostB ? "127.0.0.1", parachainPortB ? 9188, paraIdB ? 2000
```

please make one chain to be 2087(at least one of chain should be same as in testnet) and other chain to be 2137(it was our swap slot).  also please make 9188 port to be 9998(see other devnets) .

[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply code/centauri/default.nix]: Afaik @obsessed-cake works on using compose named network as defaults. He can make these changes here.

File: code/centauri/hyperspace/default.nix [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        cargoArtifacts = common-deps;
+        cargoExtraArgs = "--package hyperspace";
+        installPhase = ''
+          mkdir --parents $out/bin
+          cp target/release/hyperspace $out/bin/hyperspace
+        '';
+        meta = { mainProgram = "hyperspace"; };
+      });
+      rawConfig = (builtins.fromTOML
+        (builtins.readFile ./code/centauri/hyperspace/config.toml));
```
Comment (Wizdave97) [inline code/centauri/hyperspace/default.nix:22]: Does this setup overwrite what's in the `config.toml` file

Comment (dzmitry-lahoda) [reply code/centauri/hyperspace/default.nix]: We have to overwrite so we can target release to different hosts/ports/names/etc. So it is only builder now. Please specify any combo you need for later work.

**Review by blasrodri — APPROVED** [review]

**Pull request merged**